src/Tickets/Order_Modifiers/Checkout/Coupons.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Coupon class for the Checkout.
+ *
+ * @todo - This class is a placeholder, and will need to be refactored/optimized when Coupons are worked on.
+ */
+
+namespace TEC\Tickets\Order_Modifiers\Checkout;
+
+use TEC\Common\StellarWP\Assets\Asset;
+use TEC\Tickets\Order_Modifiers\Modifiers\Coupon;
+use TEC\Tickets\Registerable;
+use Tribe__Assets;
+use Tribe__Template;
+use WP_Post;
+
+/**
+ * Class Coupons
+ *
+ * Handles coupon logic in the checkout process.
+ *
+ * @since TBD
+ */
+class Coupons implements Registerable {
+
+	/**
+	 * @var Coupon
+	 */
+	protected Coupon $coupon;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		$this->coupon = new Coupon();
+	}
+
+	/**
+	 * Registers hooks and AJAX actions.
+	 *
+	 * @since TBD
+	 */
+	public function register(): void {
+		// Hook for displaying coupons in the checkout.
+		add_action(
+			'tec_tickets_commerce_checkout_cart_before_footer_quantity',
+			[
+				$this,
+				'display_coupon_section',
+			],
+			40,
+			3
+		);
+
+		// Add asset localization to ensure the script has the necessary data.
+		add_action( 'init', fn() => $this->localize_assets() );
```

Comment:
Avoid this on production code, this hook cannot be removed.

Reply:
> this hook cannot be removed.

Making the hook difficult to remove is intentional. Many of our hooks aren't meant to be removed. If there's a reason we might want to allow certain hooks to be easily removed or modified, then it makes sense to change how they are hooked so that they can be removed more easily.

Technically speaking, these *can* still be removed, but it is difficult to do so with the current implementation of the WP Hooks API.